src/screens/DiscoveryScreenV2/elements/RecentSearch.js

```diff
@@ -20,106 +20,100 @@ import {COLORS} from '../../../utils/theme';
-    };
-  }, [shown]);
-
-  React.useEffect(() => {
-    setItems(discovery.recentSearch);
-    return () => {
-      setItems([]);
-    };
-  }, [discovery.recentSearch]);
-
-  const manipulateSearchTermsOrder = async (search) => {
-    const result = await AsyncStorage.getItem(RECENT_SEARCH_TERMS);
-    if (!result) return;
-
-    let resultArray = JSON.parse(result);
-    if (resultArray.length <= 1) return;
-
-    const itemIndex = resultArray.indexOf(search);
-    resultArray = [search].concat(resultArray);
-    resultArray.splice(itemIndex + 1, 1);
-    DiscoveryAction.setDiscoveryRecentSearch(resultArray, discoveryDispatch);
-    AsyncStorage.setItem(RECENT_SEARCH_TERMS, JSON.stringify(resultArray));
-    setItems(resultArray);
-  };
-
-  const fetchDiscoveryData = async (search) => {
-    setSearchText(search);
-    setIsFirstTimeOpen(false);
-    Keyboard.dismiss();
-    manipulateSearchTermsOrder(search);
-  };
-
-  const handleClearRecentSearch = async () => {
-    setIsShown(false);
-    // Remove from context here
-    await AsyncStorage.removeItem(RECENT_SEARCH_TERMS);
-
-    DiscoveryAction.setDiscoveryRecentSearch([], discoveryDispatch);
-  };
-
-  if (isShown && items.length > 0)
-    return (
-      <View style={styles.recentContainer}>
+    const {
+        shown = true,
+        setSearchText = () => { },
+        setIsFirstTimeOpen = () => { }
+    } = param
+
+    const [discovery, discoveryDispatch] = React.useContext(Context).discovery
+
+    const [isShown, setIsShown] = React.useState(shown)
+    const [items, setItems] = React.useState(discovery.recentSearch)
+
+    React.useEffect(() => {
+        setIsShown(shown)
+        return () => {
+            setIsShown(false)
+        }
+    }, [shown])
+
+    React.useEffect(() => {
+        setItems(discovery.recentSearch)
+        return () => {
+            setItems([])
+        }
+    }, [discovery.recentSearch])
+
+    const manipulateSearchTermsOrder = async (search) => {
+        const result = await AsyncStorage.getItem(RECENT_SEARCH_TERMS)
+        if (!result) return
+
+        let resultArray = JSON.parse(result)
+        if (resultArray.length <= 1) return
+
+        const itemIndex = resultArray.indexOf(search)
+        resultArray = [search].concat(resultArray)
+        resultArray.splice(itemIndex + 1, 1)
+        DiscoveryAction.setDiscoveryRecentSearch(resultArray, discoveryDispatch)
+        AsyncStorage.setItem(RECENT_SEARCH_TERMS, JSON.stringify(resultArray))
+        setItems(resultArray)
+
+    }
+
+    const fetchDiscoveryData = async (search) => {
+        setSearchText(search)
+        setIsFirstTimeOpen(false)
+        Keyboard.dismiss()
+        manipulateSearchTermsOrder(search)
+    }
+
+    const handleClearRecentSearch = async () => {
+        setIsShown(false)
+        // Remove from context here
+        await AsyncStorage.removeItem(RECENT_SEARCH_TERMS)
+
+        DiscoveryAction.setDiscoveryRecentSearch([], discoveryDispatch)
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the function `manipulateSearchTermsOrder`, there is a potential issue with the manipulation of `resultArray`. The item at `itemIndex` is being moved to the front of the array, but if `itemIndex` is 0 (i.e., the item is already at the front), this will result in a duplicate entry at the start of the array. To fix this, you should check if `itemIndex` is not 0 before manipulating the array.
```diff
- const itemIndex = resultArray.indexOf(search)
- resultArray = [search].concat(resultArray)
- resultArray.splice(itemIndex + 1, 1)
+ const itemIndex = resultArray.indexOf(search);
+ if (itemIndex !== 0) {
+     resultArray = [search].concat(resultArray);
+     resultArray.splice(itemIndex + 1, 1);
+ }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->